src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -106,6 +120,46 @@ public AssetProvider CreateAssetProvider(Checksum solutionChecksum, SolutionAsse
             bool updatePrimaryBranch,
             Func<Solution, ValueTask<T>> implementation,
             CancellationToken cancellationToken)
+        {
+            try
+            {
+                // bridge from this single-value callback to the streaming callback form.
+
+                var stream = RunWithSolutionAsync(
+                    assetProvider, solutionChecksum, workspaceVersion, updatePrimaryBranch,
+                    (solution, cancellationToken) => GetStreamImplementation(solution, implementation, cancellationToken), cancellationToken);
+
+                await foreach (var pair in stream.ConfigureAwait(false))
```

Comment:
would it be possible to convert to a regular enumerable and then assert that we got a single result and return that?

Reply:
hrmm... i'm not really seeing how we could do that.  in that the 'stream' of data being passed in is itself an IAsyncEnumerable.  We could try to *realize* that into a full list.  but hten you lose the end-to-end streaming from the searching all the way to the client pulling on things.

Reply:
I think I mean something like `stream.ToListAsync()` then asserting that there's only a single result.  Though now that I look that might be in a nuget package we aren't using..

> but hten you lose the end-to-end streaming
in this method though it looks like we're only returning a single result, so we don't need to stream anything.  But I might be missing something.

Reply:
ah, i see what you mean.  i'll do that :)

Reply:
done.